lib/main.js

```diff
@@ -385,54 +387,59 @@ function createPipelinedRequest(cb){
  * @param options Options for the request function
  * @param cb  Callback to handle response
  * */
-function request(options,cb){
-    var _options={};
-    if(isThrottled()){
-        var host=options.host;
-        var port=options.port;
-        var path=options.path;
-        var method=options.method;
-        var proxyPort=process.pid%60000;
-        _options.host=localProxyHost;
-        _options.port=proxyPort;
-        _options.method=method;
-        if(options.headers !=null){
-            _options.headers=options.headers;
-        }else{
-            _options.headers={};
+function request(options, cb) {
+    var _options = {};
+    if (isThrottled()) {
+        var host = options.host;
+        var port = options.port;
+        var path = options.path;
+        var method = options.method;
+        var proxyPort = process.pid % 60000;
+        _options.host = localProxyHost;
+        _options.port = proxyPort;
+        _options.method = method;
+        if (options.headers != null) {
+            _options.headers = options.headers;
+        } else {
+            _options.headers = {};
         }
-        _options.headers['host']=host;
-        _options.headers['port']=port;
-        _options.headers['path']=path;
-        _options.headers['method']=method;
-    }else{
-        _options=options;
+        _options.headers['host'] = host;
+        _options.headers['port'] = port;
+        _options.headers['path'] = path;
+        _options.headers['method'] = method;
+    } else {
+        _options = options;
+    }
+    if (anonymizer.isAnonymizationEnabled) {
+        _options = anonymizer.anonymizeRequest(_options);
     }
     let protocol;
     if (options.port === "443" && !isThrottled()) {
         _options['agent'] = gh.keepAliveAgentHttps;
         protocol = https;
-    }
-    else {
+    } else {
         _options['agent'] = gh.keepAliveAgentHttp;
         protocol = http;
     }
+
     var req = protocol.request(_options, function (response) {
         response.on('end', function () {
             incrCounter(response.statusCode);
         });
-
-        if(cb){
-            cb(response);
+    
+        if (cb) {
+            if(anonymizer.isAnonymizationEnabled)
+            anonymizer.anonymizeResponse(response, cb);
+        } else {
+            response.resume();
```

Comment:
@AJun01 You have missed the cb(response) on the flow where anonymization is not enabled. Additionally, always use { } in if statements, even if it is one liner

Reply:
I would expect to see something like this, based on your code:

```
if (cb) {
  if(anonymizer.isAnonymizationEnabled) {
    anonymizer.anonymizeResponse(response, cb);
   }else {
     cb(response)
   }
}
response.resume();
```